server/sonar-server/src/main/java/org/sonar/server/permission/PermissionFinder.java

```diff
@@ -19,141 +19,75 @@
  */
 package org.sonar.server.permission;
 
-import com.google.common.base.Predicate;
-import com.google.common.collect.Iterables;
-import java.util.Collection;
+import com.google.common.collect.Ordering;
 import java.util.List;
-import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
+import java.util.stream.Collectors;
 import org.sonar.api.security.DefaultGroups;
 import org.sonar.api.server.ServerSide;
-import org.sonar.api.utils.Paging;
-import org.sonar.core.permission.GlobalPermissions;
 import org.sonar.db.DbClient;
 import org.sonar.db.DbSession;
-import org.sonar.db.component.ResourceDao;
-import org.sonar.db.component.ResourceDto;
-import org.sonar.db.component.ResourceQuery;
-import org.sonar.db.permission.GroupWithPermissionDto;
 import org.sonar.db.permission.PermissionDao;
 import org.sonar.db.permission.PermissionQuery;
-import org.sonar.db.permission.UserWithPermissionDto;
-import org.sonar.server.exceptions.NotFoundException;
+import org.sonar.db.user.GroupDao;
+import org.sonar.db.user.GroupDto;
+import org.sonar.db.user.GroupRoleDto;
+import org.sonar.db.user.UserDao;
+import org.sonar.db.user.UserDto;
+import org.sonar.db.user.UserPermissionDto;
 
-import static com.google.common.collect.Lists.newArrayList;
-import static org.apache.commons.lang.StringUtils.containsIgnoreCase;
-import static org.sonar.api.utils.Paging.forPageIndex;
+import static java.util.Collections.emptyList;
 
 @ServerSide
 public class PermissionFinder {
 
   private final PermissionDao permissionDao;
-  private final ResourceDao resourceDao;
+  private final UserDao userDao;
+  private final GroupDao groupDao;
 
   public PermissionFinder(DbClient dbClient) {
-    this.resourceDao = dbClient.resourceDao();
     this.permissionDao = dbClient.permissionDao();
+    this.userDao = dbClient.userDao();
+    this.groupDao = dbClient.groupDao();
   }
 
-  public List<UserWithPermissionDto> findUsersWithPermission(DbSession dbSession, PermissionQuery query) {
-    Long componentId = componentId(query.component());
-    int limit = query.pageSize();
-    return permissionDao.selectUsers(dbSession, query, componentId, offset(query), limit);
-  }
+  public List<UserDto> findUsers(DbSession dbSession, PermissionQuery.Builder dbQuery) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)